ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go

```diff
@@ -501,11 +501,19 @@ func (h *FaultHandler) CheckNetworkPacketLoss() func(http.ResponseWriter, *http.
 	}
 }
 
-// decodeRequest will translate/unmarshal an incoming fault injection request into one of the network fault structs
-func decodeRequest(w http.ResponseWriter, request types.NetworkFaultRequest, requestType string, r *http.Request) error {
+// decodeRequest will log the request and then translate/unmarshal an incoming fault injection request into
+// one of the network fault structs if the request body is required.
+func decodeRequest(w http.ResponseWriter, request types.NetworkFaultRequest, requestType string,
+	r *http.Request, requiredRequestBody bool) error {
```

Comment:
Why do we need `requiredRequestBody` parameter? If any function is calling `decodeRequest` then it should always expect a request body?

Reply:
I was thinking to decode the request for future uses (because the client can still include the request payload if that's optional). But yes you are correct. I can remove this when request body is not required.

Reply:
Removed this param.